### ISSUE TITLE
About my experiment on DANN

### ISSUE THREAD

**Issue by YuGuilliman** [Wang-ML-Lab/TSDA#5]
Hellow again!
I am now doing my experiment on DANN. In my poor knowledge, when a DANN is trained well, its domian accuracy should get lower than the beginning one, maybe retain about 60% in the end while the loss_domain should be higher. But I encounter a situation that my DANN domain accuracy suddently achieves 100% after training for 100 epochs, while the number is 50% in the past epochs(i have no idea if the number is right). I think the DANN should not work well in such a situatiion, but the test accuracy on targrt dataset is better than before. I would like to ask if it is a right experiment or there is some other factors so the tgt_acc gets higer. For details, I use Alexnet as the backboone, both the domain discriminator and the classifier consist of 3 linear layers, and I set the weight for the discriminator loss to 1 in the total loss computation.
Looking forward for your kind help!
(if every thing works well there should be 2 figures bellow, and eval_acc=tgt_acc)
![b497b63618df0d69f800177c8439aac](https://github.com/Wang-ML-Lab/TSDA/assets/79433082/2ba4ed79-a568-43bc-9fac-4a16deb278d2)
![c80a0f455d864ddd339d123b68e84b4](https://github.com/Wang-ML-Lab/TSDA/assets/79433082/aa14301e-ade9-4d5a-8927-25d26b13ec48)

**Comment by shsjxzh** [comment]
Hi YuGuilliman,

It is possible that DANN occasionally performs quite well. This might be due to DANN finding an optimal solution under certain hyperparameters. To conduct a thorough comparison, you could run experiments using several random seeds and compare both the mean and the variance.

**Comment by YuGuilliman** [comment]
I get your idea, thank you for your kind help!
